### PR TITLE
URL parsing error when the URL has port.

### PR DESCRIPTION
**Pull request by gmarcos87**
URL parsing error when the URL has port and hostname is the same as the site. 
Checks if the URL has port, if so adds the hostname again.

https://github.com/TryGhost/Ghost/issues/6893
Test OK and Test case Ok.

### CODE REVIEW

**Comment by ErisDS** [comment]
Hi there @gmarcos87, thanks so much for taking the time to contribute to Ghost and sorry for taking a while to review your PR.

When fixing bugs like this, we ask that you also add a test which would break before your fix and which works with your fix applied, so we can see that your code does fix the bug. 

There has been another, very similar change [made recently](https://github.com/TryGhost/Ghost/pull/6945/files), you could use this as an example for how you might add tests to cover this case.

Thanks 👍

**Comment by gmarcos87** [comment]
Hi Eris, 
I included the test now

Thanks

File: core/server/config/url.js [hunk 1]
```diff
@@ -211,7 +211,11 @@ function urlFor(context, data, absolute) {
...
                 urlPath = urlPath.split(hostname)[1];
                 if (urlPath.substring(0, 1) !== '/') {
-                    urlPath = '/' + urlPath;
+                    if (urlPath.substring(0, 1) !== ':') {
+                        urlPath = '/' + urlPath;
+                    } else {
+                        urlPath = data.nav.url;
+                    }
                 }
                 absolute = true;
```
Comment (kirrg001) [inline core/server/config/url.js]: this flag here should be false in the case you have added.

File: core/test/unit/config_spec.js [hunk 2]
```diff
@@ -377,6 +377,11 @@ describe('Config', function () {
                 testData = {nav: {url: 'http://my-ghost-blog.com/blog/short-and-sweet/'}};
                 config.urlFor(testContext, testData).should.equal('http://my-ghost-blog.com/blog/short-and-sweet/');
 
+                configUtils.set({url: 'http://my-ghost-blog.com'});
+                testData = {nav: {url: 'http://my-ghost-blog.com:3000/short-and-sweet/'}};
+                config.urlFor(testContext, testData).should.equal('http://my-ghost-blog.com:3000/short-and-sweet/');
+                config.urlFor(testContext, testData).should.not.equal('http://my-ghost-blog.com/:3000/short-and-sweet/');
```
Comment (kirrg001) [inline core/test/unit/config_spec.js]: you can keep this line, but this line is not really needed - you already assert the correct result in line `382`.

File: core/test/unit/config_spec.js [hunk 3]
```diff
@@ -377,6 +377,11 @@ describe('Config', function () {
                 testData = {nav: {url: 'http://my-ghost-blog.com/blog/short-and-sweet/'}};
                 config.urlFor(testContext, testData).should.equal('http://my-ghost-blog.com/blog/short-and-sweet/');
 
+                configUtils.set({url: 'http://my-ghost-blog.com'});
+                testData = {nav: {url: 'http://my-ghost-blog.com:3000/short-and-sweet/'}};
```
Comment (kirrg001) [inline core/test/unit/config_spec.js]: Maybe another edge case we can add here:
`my-ghost-blog.com:3000/short-and-sweet/`

**Comment by kirrg001** [comment]
If you're interested: somebody else also created a PR for this bug
https://github.com/TryGhost/Ghost/pull/6998/files

Thanks for contributing 👍

**Comment by gmarcos87** [comment]
That is better. Sorry for the lost time is very first contribution

**Comment by kirrg001** [comment]
@gmarcos87 No problem, we are happy about any help!
Tell me what do you prefer: updating your PR or merging #6998 - up to you :)

**Comment by kirrg001** [comment]
@gmarcos87 ?

**Comment by kevinansfield** [comment]
Closing in favour of #6998.

**Pull request closed without merging**